### PR TITLE
Add more C++ tests for activations

### PR DESCRIPTION
**Pull request by ptrendx**
# Description

This PR consolidates tests for GLU and regular versions of the activations as well as improves coverage of the activations (earlier mostly covering GELU/GEGLU).

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Please list the changes introduced in this PR:

- Added coverage of relu/silu/reglu/swiglu in C++ tests
- Combined test_gelu.cu and test_geglu.cu into a single test_act.cu file

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/cpp/operator/test_act.cu [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+float gelu(const float x) {
+    return 0.5f * x * (1.0f + tanhf(0.79788456F * x * (1.0f + 0.044715f * x * x)));
+}
+
+float silu(const float x) {
+  return x / (1 + expf(-x));
+}
+
+float relu(const float x) {
```
Comment (phu0ngng) [inline tests/cpp/operator/test_act.cu:34]: Why don't we have `srelu` and `qgelu`?

**Review by phu0ngng — APPROVED** [review]

**Review by timmoon10 — APPROVED** [review]
LGTM

File: tests/cpp/operator/test_act.cu (hunk 1 above)
Comment (ptrendx) [reply tests/cpp/operator/test_act.cu:34]: Good point.

**Comment by phu0ngng** [comment]
LGTM!

**Pull request merged**